gqcp/tests/Basis/Transformations/OrbitalRotationGenerators_test.cpp

```diff
@@ -19,24 +19,98 @@
 
 #include <boost/test/unit_test.hpp>
 
-#include "Basis/Transformations/OrbitalRotationGenerators.hpp"
+#include "Basis/Transformations/GOrbitalRotationGenerators.hpp"
+#include "Basis/Transformations/ROrbitalRotationGenerators.hpp"
+#include "Basis/Transformations/UOrbitalRotationGenerators.hpp"
+#include "Basis/Transformations/UOrbitalRotationGeneratorsComponent.hpp"
 
 
 /**
- *  Check the OrbitalRotationGenerators constructor.
+ *  Check the ROrbitalRotationGenerators constructor.
  */
-BOOST_AUTO_TEST_CASE(constructor) {
+BOOST_AUTO_TEST_CASE(restrictedConstructor) {
 
     // Check if the constructor throws upon receiving an argument of wrong dimensions.
     const GQCP::VectorX<double> kappa {4};  // '4' is not a triangular number.
-    BOOST_CHECK_THROW(GQCP::OrbitalRotationGenerators generators {kappa}, std::invalid_argument);
+    BOOST_CHECK_THROW(GQCP::ROrbitalRotationGenerators<double> generators {kappa}, std::invalid_argument);
+}
+
+/**
+ *  Check the UOrbitalRotationGeneratorsComponent constructor.
+ */
+BOOST_AUTO_TEST_CASE(unrestrictedComponentConstructor) {
+
+    // Check if the constructor throws upon receiving an argument of wrong dimensions.
+    const GQCP::VectorX<double> kappa {4};  // '4' is not a triangular number.
+    BOOST_CHECK_THROW(GQCP::UOrbitalRotationGeneratorsComponent<double> generators {kappa}, std::invalid_argument);
+}
+
+/**
+ *  Check the GOrbitalRotationGenerators constructor.
+ */
+BOOST_AUTO_TEST_CASE(generalizedConstructor) {
+
+    // Check if the constructor throws upon receiving an argument of wrong dimensions.
+    const GQCP::VectorX<double> kappa {4};  // '4' is not a triangular number.
+    BOOST_CHECK_THROW(GQCP::GOrbitalRotationGenerators<double> generators {kappa}, std::invalid_argument);
+}
```

Comment:
In my opinion, since you're really just testing `SimpleOrbitalRotationGenerators` functionality, you don't have to write unit tests for each of the deriving classes. So I'd just keep one of the cases, unless it's specific functionality of one of the child classes that you're trying to test.